core/server/worker/src/main/java/alluxio/worker/grpc/GrpcExecutors.java

```diff
@@ -54,6 +54,12 @@
   public static final ExecutorService BLOCK_READER_EXECUTOR =
       new ImpersonateThreadPoolExecutor(BLOCK_READER_THREAD_POOL_EXECUTOR);
 
+  public static final ExecutorService BLOCK_MOVER_EXECUTOR =
+      new ImpersonateThreadPoolExecutor(new ThreadPoolExecutor(THREADS_MIN,
+          ServerConfiguration.getInt(PropertyKey.WORKER_NETWORK_BLOCK_MOVER_THREADS_MAX),
+          THREAD_STOP_MS, TimeUnit.MILLISECONDS, new SynchronousQueue<>(),
```

Comment:
Considering we will have a large volume or read requests typically, I think we should not use a queue with no length limit. Also we should discuss on what the proper rejection policy is, maybe just throw the request away without exception (but increment a metric and do sample logging).
What do you think? @beinan @apc999

Reply:
Good call!  adding metrics and logging sounds great!

Reply:
Sounds Great! I will add a rejection policy with failure metrics and loggings.